### PR TITLE
Simplify logic behind fa(), update script, and reduce size of saved data

### PR DESCRIPTION
**Pull request by cpsievert**

### CODE REVIEW

**Comment by wch** [comment]
Just a heads up: some of the things in the existing code that look unusual were done deliberately. The data storage format, for example: see #48. Also, some (not necessarily all) of the unusual code in `fa()` was done for performance reasons. If this code is going to be refactored, we should make sure that it's done with this in mind.

**Comment by cpsievert** [comment]
Thanks @wch, I did sacrifice some readability for performance in https://github.com/rstudio/fontawesome/pull/79/commits/acc73757c2659b30d61a48020dc9bfb53e8fd4d2 and now performance is about the same

Before

```R
> system.time(for (i in 1:5000) as.character(fontawesome::fa("calendar")))
   user  system elapsed 
  0.564   0.037   0.602 
```

After 

```R
> system.time(for (i in 1:5000) as.character(fontawesome::fa("calendar")))
   user  system elapsed 
  0.644   0.024   0.676
```


As for getting diffs in the icon table, this PR will now save a version of the table under `data-raw/fa_tbl.csv`. And so, since we can track diffs via that file, I figured it was OK to revert back to the `.rda` approach  for `fa_tbl` (instead of embedding the data in a string under `R/`), but if there are other reasons to encode the data that way, it'd be good to know what they are

File: R/fa_html_dependency.R [hunk 1]
```diff
@@ -36,5 +34,11 @@ fa_dependency_obj <- NULL
 #'
 #' @export
 fa_html_dependency <- function() {
-  fa_dependency_obj
+  htmltools::htmlDependency(
```
Comment (wch) [inline R/fa_html_dependency.R:37]: Should pre-generate this and move back into `.onLoad()`.

**Comment by cpsievert** [comment]
@rich-iannone I'm pretty sure the routine won't run since I don't  have write/admin privileges. Once I'm added we can look into that in another PR.

**Review by rich-iannone — APPROVED** [review]
Everything looks good and manual testing (using Shiny testing apps and R Markdown renders) show no apparent changes from `main`.

**Pull request merged**